### PR TITLE
Update MPC Contract Signature Arguments

### PR DESCRIPTION
**Pull request by bh2smith**
The Near MPC Contract was recently updated (the whole interface has been modified) - specifically the arguments for the sign method: See here https://github.com/near/mpc-recovery/pull/613

This contract is deployed to: https://testnet.nearblocks.io/address/v1.signer-prod.testnet

In this transaction: https://testnet.nearblocks.io/txns/E2fQfcdqPARCnok7RFAJ6WXur3SDKEB7ESgYiH5zHfBT

**This PR**
1. Adopts the new MPC contract sign method interface.
2. Lowers the default gas limit on the signature request (cf https://github.com/near/mpc-recovery/pull/589)
3. Introduces a deposit of 1 YOCTO as is now required.
4. Updates the kdf (key derivation) to use sha3 instead of sha2 as outlined in https://github.com/near/mpc/blob/develop/CHANGELOG.md
5. Payloads are now all big endian instead of little endian (so no longer need to be reversed).

Note that the recovery ID is now returned with the MPC Signature, so we can remove A LOT of fluff related to signature "Recovery". There may still be more fluff that can be removed (or perhaps some may need to be reintroduced). Unfortunately the availability of the signing service is not stable enough to test everything. However, we have seen a successful request already: https://sepolia.etherscan.io/tx/0x1adfbaa3151ebdd69b790da41e719eaa039861bd7f56e34fd9f280bded63f535

This change is "breaking" in the sense that the MPC contract interface no longer accets the old Near Tx Payload. This means that the project will no longer work for users pointing at the [v2.multichain-mpc.testnet](https://testnet.nearblocks.io/address/v2.multichain-mpc.testnet) as it does not appear to have been updated along with [v1.signer-prod.testnet](https://testnet.nearblocks.io/address/v1.signer-prod.testnet) which we use by default here.

### CODE REVIEW

**Comment by DavidM-D** [comment]
[Here's the full changelog](https://github.com/near/mpc/blob/develop/CHANGELOG.md) for the v5 contract.
Broadly, don't reverse the hash scalar any more, don't reverse the key derivation scalar and use sha3 instead of sha2 for the key derivation. Happy to review, or we'll do it later this week.

**Comment by DavidM-D** [comment]
[Here's the full changelog](https://github.com/near/mpc/blob/develop/CHANGELOG.md) for the v5 contract.
Broadly, don't reverse the hash scalar any more, don't reverse the key derivation scalar and use sha3 instead of sha2 for the key derivation. Happy to review, or we'll do it later this week on our libs.

File: src/utils/kdf.ts [hunk 1]
```diff
@@ -48,11 +48,5 @@ export function uncompressedHexPointToEvmAddress(
...
   const encoder = new TextEncoder();
   const data = encoder.encode(str);
-  const hashBuffer = await crypto.subtle.digest("SHA-256", data);
-  const hashArray = [...new Uint8Array(hashBuffer)];
-  return hashArray.map((b) => b.toString(16).padStart(2, "0")).join("");
-}
-
-function sha256StringToScalarLittleEndian(hashString: string): string {
-  return hashString.match(/../g)!.reverse().join("");
+  return keccak256(data, "hex");
```
Comment (bh2smith) [inline src/utils/kdf.ts:51]: Maybe this wasn't prefixed with 0x before and now is...!

**Comment by bh2smith** [comment]
Well that was a nightmare. BUT I GOT IT!

**Pull request merged**